src/Microsoft.AspNet.FileProviders.Embedded/EmbeddedFileProvider.cs

```diff
@@ -142,6 +142,15 @@ public IChangeToken Watch(string pattern)
             return NoopChangeToken.Singleton;
         }
 
+        private static bool HasInvalidPathChars(string path)
+        {
+            // convert path separators to a safe character so we can check for invalid file name characters
+            // and not get false positives from '/' or '\\'
+            var subPath = path.Replace(Path.DirectorySeparatorChar, '.')
```

Comment:
Could we rewrite this method to avoid performing these `Replace`s? Just avoiding the extra allocations.